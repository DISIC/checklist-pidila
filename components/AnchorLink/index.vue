<template>
    <a tabindex="0" @click="scrollTo(anchor)" @keyup.enter="scrollTo(anchor)" :class="cssClass">
      {{message}}
    </a>
</template>

<script>
  export default {
    name: 'AnchorLink',
    props: {
      anchor: {
        type: String,
        default: ''
      },
      cssClass: {
        type: String,
        default: ''
      },
      message: {
        type: String,
        default: ''
      }
    },
    methods: {
      scrollTo: function (anchor) {
        this.$router.scrollTo(anchor)
      }
    }
  }
</script>
