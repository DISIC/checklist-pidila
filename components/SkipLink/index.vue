<template>
  <div class="skip-link">
    <div class="container" id="top">
      <ul>
        <li><anchor-link :anchor="'main'" :message="'Aller au contenu'"/></li>
        <li><anchor-link :anchor="'filters'" :message="'Aller aux filtres'"/></li>
      </ul>
    </div>
  </div>
</template>

<script>
  import AnchorLink from '../AnchorLink/index'

  export default {
    components: {AnchorLink},
    name: 'SkipLink',
    methods: {
      scrollTo: function (anchor) {
        this.$router.scrollTo(anchor)
      }
    }
  }
</script>
