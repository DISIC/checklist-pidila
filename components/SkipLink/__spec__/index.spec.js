import {shallowMount} from '@vue/test-utils'
import SkipLink from '../index.vue'
import AnchorLink from '../../AnchorLink'

describe('components | SkipLink', () => {
  it('should match snapshot', () => {
    // when
    const wrapper = shallowMount(SkipLink)

    // then
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should have two AnchorLink components to #main and #filters', () => {
    // when
    const wrapper = shallowMount(SkipLink)
    const expectedProps = [
      {anchor: 'main', cssClass: '', message: 'Aller au contenu'},
      {anchor: 'filters', cssClass: '', message: 'Aller aux filtres'}
    ]

    // then
    const anchorLinkComponents = wrapper.findAll(AnchorLink)
    expect(anchorLinkComponents.length).toEqual(2)

    anchorLinkComponents.wrappers.forEach((component, index) => {
      expect(component.props()).toEqual(expectedProps[index])
    })
  })
})
