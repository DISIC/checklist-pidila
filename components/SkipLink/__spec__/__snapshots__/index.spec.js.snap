// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | SkipLink should match snapshot 1`] = `
<div
  class="skip-link"
>
  <div
    class="container"
    id="top"
  >
    <ul>
      <li>
        <anchor-link-stub
          anchor="main"
          cssclass=""
          message="Aller au contenu"
        />
      </li>
       
      <li>
        <anchor-link-stub
          anchor="filters"
          cssclass=""
          message="Aller aux filtres"
        />
      </li>
    </ul>
  </div>
</div>
`;
